streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -240,8 +240,8 @@
                     STATE_DIR_DOC)
             .define(REPLICATION_FACTOR_CONFIG,
                     Type.INT,
-                    1,
-                    Importance.MEDIUM,
+                    3,
```

Comment:
yeah, my suggestion was to make the dynamism configurable. 

REPLICATION_FACTOR_CONFIG= N where N >0 => i know what i want. streams should fail if it can't meet this contract
REPLICATION_FACTOR_CONFIG = -1 => streams is smart and can figure it out for me.

this allows *me* to be as anal retentive as i want, but allows the defaults to work for demos/tests/etc.